### PR TITLE
Switch to Node 4

### LINKED ISSUE #75: Drop Node 0.12
It'd be good to use Argon (4.0) as the new minimum. Will allow some ES6 features to be used. Stylelint has already dropped it and PostCSS is due to [in the next version](https://twitter.com/PostCSS/status/833285057762697216).

### PR DESCRIPTION
**Pull request by simonsmith**
Fixes #75

### CODE REVIEW

**Comment by giuseppeg** [comment]
@simonsmith awesome! I'll take a look this week

**Comment by simonsmith** [comment]
@giuseppeg Think you can look today? I'm going to have some free time over the weekend so I'll get #72 sorted too and publish a new version

**Comment by giuseppeg** [comment]
hey @simonsmith sorry man I am afraid that I won't be able to look at it before the weekend.
Be sure though that if I manage to find five free minutes I'll take a look – I believe that it is syntax changes mostly.

**Comment by simonsmith** [comment]
@giuseppeg Yeah, most of it was just put through `lebab` and then tweaked with ESLint

**Comment by simonsmith** [comment]
In fact I can just work off this branch and merge later. So no worries

File: bin/suitcss [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-var suitcss = require('..');
-var path = require('path');
-var program = require('commander');
-var read = require('read-file-stdin');
-var chokidar = require('chokidar');
-
-var exists = fs.existsSync;
-var resolve = path.resolve;
-var writeFileSync = fs.outputFileSync;
+'use strict';
```
Comment (giuseppeg) [inline bin/suitcss]: do we need this?

File: lib/index.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+const bemLinter = require('postcss-bem-linter');
+const cssnano = require('cssnano');
+const difference = require('lodash.difference');
+const encapsulationPlugins = require('./encapsulation');
+const isEmpty = require('lodash.isempty');
+const postcssEasyImport = require('postcss-easy-import');
+const reporter = require('postcss-reporter');
+const stylelint = require('stylelint');
+const stylelintConfigSuit = require('stylelint-config-suitcss');
+let postcss = require('postcss'); // eslint-disable-line prefer-const
```
Comment (giuseppeg) [inline lib/index.js]: why `let`?

File: test/encapsulation/browser.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
-var styles = require('../fixtures/encapsulation.out.css');
+'use strict';
 
-var dom = (function () {
-  var container = document.createElement('div');
+const getStyle = require('computed-style');
+const test = require('tape-css')(require('tape'));
+const styles = require('../fixtures/encapsulation.out.css');
+
+const dom = ((() => {
```
Comment (giuseppeg) [inline test/encapsulation/browser.js]: the additional expression to wrap the iife is probably unnecessary `(() => {})()` should be enough

File: bin/suitcss (hunk 1 above)
Comment (simonsmith) [inline bin/suitcss]: Yeah, I get an error saying to use `const` and `let` it has to be in strict mode.

File: lib/index.js (hunk 2 above)
Comment (simonsmith) [inline lib/index.js]: This is just so the `rewire` library can override it in the tests

File: test/encapsulation/browser.js (hunk 3 above)
Comment (simonsmith) [inline test/encapsulation/browser.js]: Ah okay, just a copy paste from lebab. Will update

File: bin/suitcss (hunk 1 above)
Comment (giuseppeg) [inline bin/suitcss]: maybe some eslint rule that we can remove?

Comment (simonsmith) [inline bin/suitcss]: The error comes directly from Node

Comment (simonsmith) [inline bin/suitcss]:
```
SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
```

Comment (giuseppeg) [inline bin/suitcss]: same, it seems that we could pass `--use_strict` to `mocha` or `node`  but shrug lets keep `"use strict"`

Comment (simonsmith) [inline bin/suitcss]: I suppose we couldn't be sure it would be run that way by users. I noticed `stylelint` add it so probably wise to follow suit 👍

**Pull request merged**